storage/aptosdb/src/state_store/mod.rs

```diff
@@ -176,42 +176,42 @@ impl StateStore {
         Ok(result)
     }
 
+    /// Read the value for a specific version and require its existence. Call `get_value_by_value`
+    /// for most use cases.
     pub fn get_value_at_version(
         &self,
         key_and_version: &(StateKey, Version),
     ) -> Result<StateValue> {
         self.db
-            .get::<StateValueSchema>(key_and_version)?
-            .ok_or_else(|| {
-                format_err!(
-                    "Canot find value of key {:?} at version {}",
-                    key_and_version.0,
-                    key_and_version.1
-                )
+            .get::<StateValueSchema>(key_and_version)
+            .and_then(|v| {
+                v.ok_or_else(|| format_err!("State Value is missing for {:?}", *key_and_version))
             })
     }
 
-    /// Get the state value given the state key and root hash of state Merkle tree by using the
-    /// state value index. Only used for testing for now but should replace the
-    /// `get_value_with_proof_by_version` call for VM execution to fetch the value without proof.
-    #[cfg(test)]
     pub fn get_value_by_version(
         &self,
         state_key: &StateKey,
         version: Version,
     ) -> Result<Option<StateValue>> {
         let mut iter = self.db.iter::<StateValueSchema>(Default::default())?;
         iter.seek_for_prev(&(state_key.clone(), version))?;
-        Ok(iter
-            .next()
+        iter.next()
             .transpose()?
             .and_then(|((db_state_key, _), state_value)| {
                 if *state_key == db_state_key {
-                    Some(state_value)
+                    Some(Ok(state_value))
                 } else {
                     None
                 }
-            }))
+            })
+            // A hack to deal with PRE_GENESIS_VERSION
```

Comment:
This is a workaround to deal with `PRE_GENESIS_VERSION`

Reply:
This is unfortunate.. let's consider removal of support for PRE_GENESIS? :trollface:  cc @zekun000

Reply:
Hmm..this is super hacky.. and also I am worried about extra read this incurs when `get_value_by_version` is called with `StateKey` that genuinely doesn't exists in DB. Do we support `PRE_GENESIS` in prod? If not, does it makes sense to have extra read just for the tests?

Reply:
Can you add a TODO here to remove this after removing pre-genesis version?

Reply:
Can you add a TODO here to remove this after removing pre-genesis version?